server-backend/src/SwapInRunner.ts

```diff
@@ -131,23 +132,31 @@ export class SwapInRunner {
     }
 
     private async processContractFundingTx(event: NBXplorerNewTransactionEvent): Promise<void> {
+        this.logger.debug(`Event data: ${JSON.stringify(event.data)}`);
+
         const { swap } = this;
         // TODO: the output is also found by buildClaimTx(), needs refactor
         const output = event.data.outputs.find(o => o.address === swap.contractAddress);
         assert(output != null);
         const receivedAmount = new Decimal(output.value).div(1e8);
+        // Handle partial payment by checking if the received amount is different than the expected amount, if so, this is considered a failed swap but will be processed until contract is expired to be able to be refunded by the sender
         if (!receivedAmount.equals(swap.inputAmount)) {
             // eslint-disable-next-line max-len
-            this.logger.error(`Amount mismatch. Failed swap. Incoming ${receivedAmount.toNumber()}, expected ${swap.inputAmount.toNumber()} (id=${this.swap.id})`);
-            return;
-        }
-        if (this.swap.status === 'CREATED' || this.swap.status === 'CONTRACT_FUNDED_UNCONFIRMED') {
+            this.logger.warn(`Amount mismatch, partial payment. Failed swap. Incoming ${receivedAmount.toNumber()}, expected ${swap.inputAmount.toNumber()} (id=${this.swap.id})`);
+            if (this.swap.status === 'CREATED') {
```

Comment:
missing a return somewhere? otherwise i'll pass through this condition multiple times and change status twice. also, do you need to publish this status?

Reply:
True, done